### PR TITLE
Bugfix infinity-loop

### PR DESCRIPTION
**Pull request by bcarranza**
**What does it do?**
- In the new version of kubetest : https://github.com/kubeshop/testkube/releases/tag/v0.11.7 we were having an infinite loop of tests.
- Our pipeline runs never ended, despite satisfactory results.

Screen Shot 2022-03-30 at 09 33 46

- This tries to leave the responsibility to the user to enter a kubetest version and somehow freeze it to avoid failed updates. 
```

Run ManagedKube/github-action-testkube@bugfix-loop
/usr/bin/docker run --name a9e7068687344f54294ac083595969ac381_cbb234 --label 372a9e --workdir /github/workspace --rm -e AWS_REGION -e tf_version -e tg_version -e KUBERGRUNT_EXEC -e kg_version -e MONGODB_ATLAS_PUBLIC_KEY -e MONGODB_ATLAS_PRIVATE_KEY -e LOCAL_REPO_PATH -e GIT_SSH_COMMAND -e CF_API_TOKEN -e ENVIRONMENT_NAME -e TESTKUBE_VERSION -e tf_working_dir -e AWS_DEFAULT_REGION -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY -e AWS_SESSION_TOKEN -e INPUT_TEST-SUITE-NAME -e INPUT_CLUSTER-NAME -e INPUT_TESTKUBE-VERSION -e INPUT_ENABLE-DEBUG -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e ACTIONS_ID_TOKEN_REQUEST_URL -e ACTIONS_ID_TOKEN_REQUEST_TOKEN -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/gruntwork-infrastructure-live/gruntwork-infrastructure-live":"/github/workspace" 372a9e:7068687344f54294ac083595969ac381  "infra-base" "false" "ops" "0.11.5"
Installing testkube specific version
Getting kubectl-testkube plugin
Downloading testkube from URL: https://github.com/kubeshop/testkube/releases/download/v0.11.5/testkube_0.11.5_Linux_x86_64.tar.gz
kubectl-testkube installed in /usr/local/bin/kubectl-testkube
Running TestSuite: infra-base
Added new context arn:aws:eks:us-west-2:***:cluster/ops to /github/home/.kube/config
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:58:47Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.15-eks-9c63c4", GitCommit:"9c63c4037a56f9cad887ee76d55142abd4155179", GitTreeState:"clean", BuildDate:"2021-10-20T00:21:03Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.19) exceeds the supported minor version skew of +/-1


████████ ███████ ███████ ████████ ██   ██ ██    ██ ██████  ███████ 
   ██    ██      ██         ██    ██  ██  ██    ██ ██   ██ ██      
   ██    █████   ███████    ██    █████   ██    ██ ██████  █████   
   ██    ██           ██    ██    ██  ██  ██    ██ ██   ██ ██      
   ██    ███████ ███████    ██    ██   ██  ██████  ██████  ███████ 
                                           /tɛst kjub/ by Kubeshop


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 144.807537ms

  STATUS  | STEP                                              | ID | ERROR  
+---------+---------------------------------------------------+----+-------+
  pending | run:testkube/infra-base-prometheus-endpoint       |    |        
  queued  | run:testkube/infra-base-alertmanager-slack-config |    |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 1.240351844s

  STATUS  | STEP                                              | ID | ERROR  
+---------+---------------------------------------------------+----+-------+
  pending | run:testkube/infra-base-prometheus-endpoint       |    |        
  queued  | run:testkube/infra-base-alertmanager-slack-config |    |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 2.238964237s

  STATUS  | STEP                                              | ID | ERROR  
+---------+---------------------------------------------------+----+-------+
  pending | run:testkube/infra-base-prometheus-endpoint       |    |        
  queued  | run:testkube/infra-base-alertmanager-slack-config |    |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 3.243909963s

  STATUS  | STEP                                              | ID                       | ERROR  
+---------+---------------------------------------------------+--------------------------+-------+
  success | run:testkube/infra-base-prometheus-endpoint       | 6243e6ff7ae8b7cd9cc210ff |        
  pending | run:testkube/infra-base-alertmanager-slack-config |                          |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 4.235944221s

  STATUS  | STEP                                              | ID                       | ERROR  
+---------+---------------------------------------------------+--------------------------+-------+
  success | run:testkube/infra-base-prometheus-endpoint       | 6243e6ff7ae8b7cd9cc210ff |        
  pending | run:testkube/infra-base-alertmanager-slack-config |                          |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 5.240640587s

  STATUS  | STEP                                              | ID                       | ERROR  
+---------+---------------------------------------------------+--------------------------+-------+
  success | run:testkube/infra-base-prometheus-endpoint       | 6243e6ff7ae8b7cd9cc210ff |        
  pending | run:testkube/infra-base-alertmanager-slack-config |                          |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : pending
Duration: 6.237[29](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:29)8657s

  STATUS  | STEP                                              | ID                       | ERROR  
+---------+---------------------------------------------------+--------------------------+-------+
  success | run:testkube/infra-base-prometheus-endpoint       | 6243e6ff7ae8b7cd9cc210ff |        
  pending | run:testkube/infra-base-alertmanager-slack-config |                          |        


Name          : infra-base
Execution ID  : 6243e6ff7ae8b7cd9cc210fd
Execution name: infra-base.briefly-flowing-bluegill
Status        : success
Duration: 6.[34](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:34)7s

  STATUS  | STEP                                              | ID                       | ERROR  
+---------+---------------------------------------------------+--------------------------+-------+
  success | run:testkube/infra-base-prometheus-endpoint       | 6243e6ff7ae8b7cd9cc210ff |        
  success | run:testkube/infra-base-alertmanager-slack-config | 6243e70[37](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:37)ae8b7cd9cc21101 |        


Test Suite execution completed with sucess in 6.347020221s 🥇

Use following command to get test execution details:
$ kubectl testkube get tse 62[43](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:43)e6ff7ae8b7cd9cc210fd


LAST_TEST_STATUS: success
LAST_TEST_EXECUTION_LIST: [ { "id": "[62](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:62)43e6ff7ae8b7cd9cc210ff", "name": "run:testkube/infra-base-prometheus-endpoint", "status": "success", "type": "executeTest" }, { "id": "6243e[70](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:70)37ae8b7cd9cc21[101](https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5748683192?check_suite_focus=true#step:13:101)", "name": "run:testkube/infra-base-alertmanager-slack-config", "status": "success", "type": "executeTest" } ]
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,9 +9,6 @@ RUN echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | tee -a /etc/a
 RUN apt-get update
 RUN apt-get install -y kubectl
 
-# Install testkube plugin
-RUN curl -sSLf https://kubeshop.github.io/testkube/install.sh | bash
-
```
Comment (bcarranza) [inline Dockerfile:14]: It is removed from the dockerfile, since it is not possible to send input variables at this point.

File: action.yml [hunk 2]
```diff
@@ -13,6 +13,10 @@ inputs:
     description: "Cluster name of test suite is hosted"
     required: true
     default: "env-name"
+  testkube-version:
+    description: "Specific version of https://github.com/kubeshop/testkube/releases"
+    required: true
+    default: "v0.11.5"
```
Comment (bcarranza) [inline action.yml]: new input var called: testkube-version

File: entrypoint.sh [hunk 3]
```diff
@@ -3,14 +3,17 @@
 TEST_SUITE_NAME=$1
 ENABLE_DEBUG=$2
 CLUSTER_NAME=$3
-
+TESTKUBE_VERSION=$4
```
Comment (bcarranza) [inline entrypoint.sh]: New envar According to the installation file: https://kubeshop.github.io/testkube/install.sh, it gives us the option to specify the version we need.

File: entrypoint.sh [hunk 4]
```diff
@@ -3,14 +3,17 @@
...
 set -e
 
 # Enable debugging output
 if [[ "${ENABLE_DEBUG}" == "true" ]]; then
     set -x
 fi
 
+# Install testkube plugin
+echo "Installing testkube specific version"
+curl -sSLf https://kubeshop.github.io/testkube/install.sh | bash
```
Comment (bcarranza) [inline entrypoint.sh]: The testkube is installed on the entry point side with the specific version defined in the env var: $TESTKUBE_VERSION

Comment (sekka1) [reply entrypoint.sh]: While it is nice to be able to define the version on the fly for each run it is not so nice that you have to download this binary and run the install every single time the action runs.  

Please put this download logic into the Dockerfile.

While you can't take user input params for the Dockerfile, peg the version there and we will do releases for the versions we want.

Comment (bcarranza) [reply entrypoint.sh:16]: Hi @sekka1 , one question: are you ok with the version number being hardcoded in the dockerfile, as I don't think I have a way to get an input variable to that point. or I have?

File: Dockerfile [hunk 5]
```diff
@@ -9,7 +9,8 @@ RUN echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | tee -a /etc/a
 RUN apt-get update
 RUN apt-get install -y kubectl
 
-# Install testkube plugin
+# Install testkube plugin - Specific version of https://github.com/kubeshop/testkube/releases
+ENV TESTKUBE_VERSION=0.11.5
 RUN curl -sSLf https://kubeshop.github.io/testkube/install.sh | bash
```
Comment (bcarranza) [inline Dockerfile:14]: Specific version of testkube

File: entrypoint.sh (hunk 4 above)
Comment (bcarranza) [reply entrypoint.sh]: Hi @sekka1 , your request was attended 👍 
new workflow run of evidence proof: https://github.com/exact-payments/gruntwork-infrastructure-live/runs/5758570897?check_suite_focus=true#step:13:25

**Review by sekka1 — APPROVED** [review]
lgtm

**Pull request merged**